### PR TITLE
Update CA2000 to exclude some types

### LINKED ISSUE #3085: Feature Request: IDE0067/CA2000 - Exclude only certain types
I think both IDE0067 and CA2000 are too valuable to turn off.

However, sometimes types that don't really have unmanaged ressources implement IDisposable for whatever reasons. I cant change those classes, they are part of a library. I can exclude them but manually setting pragmas around hundreds and more cases is really detrimental to the readability of the code.

**My wish:**
Being able to configure IDE0067 and CA2000 to exclude types, like they exclude Task.

PS:
The type in question is HalconDotNet.HTuple. Pretty niche, i know.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3085

### CODE REVIEW

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.cs [hunk 1]
```diff
@@ -62,6 +62,7 @@ private DisposeAnalysis(DisposeAnalysisDomain analysisDomain, DisposeDataFlowOpe
                 disposeOwnershipTransferLikelyTypes, disposeOwnershipTransferAtConstructor,
                 disposeOwnershipTransferAtMethodCall, trackInstanceFields, exceptionPathsAnalysis,
                 performCopyAnalysis: analyzerOptions.GetCopyAnalysisOption(rule, defaultValue: performCopyAnalysisIfNotUserConfigured, cancellationToken),
+                excludedSymbols: analyzerOptions.GetExcludedSymbolNamesOption(rule, wellKnownTypeProvider.Compilation, cancellationToken),
```
Comment (Evangelink) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.cs]: This is the "deepest" location where we still have access to the diagnostic.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeObjectsBeforeLosingScopeTests.cs [hunk 2]
```diff
@@ -11774,5 +11774,88 @@ class Test
     }
 }");
         }
+
+        [Fact, WorkItem(3085, "https://github.com/dotnet/roslyn-analyzers/issues/3085")]
+        public async Task LocalInvocationOfAnExcludedType_NoDiagnostic()
```
Comment (Evangelink) [inline src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeObjectsBeforeLosingScopeTests.cs]: Do we want more tests for the syntax of `excluded_symbol_names`?

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs [hunk 3]
```diff
@@ -97,6 +97,12 @@ protected override DisposeAbstractValue HandleInstanceCreation(IOperation creati
                     return defaultValue;
                 }
 
+                // Handle user option for additional excluded types
+                if (DataFlowAnalysisContext.ExcludedSymbols.Contains(instanceType))
```
Comment (Evangelink) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs]: I am only handling fully qualified types and type names, let me know if you think that's worth also handling namespaces.

Comment (mavasani) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs]: I think this is fine.

File: src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeObjectsBeforeLosingScopeTests.cs (hunk 2 above)
Comment (mavasani) [reply src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeObjectsBeforeLosingScopeTests.cs]: I think one test is good enough here.

**Pull request merged**